test/tests_for_sdkmr/ee2_SDKMethodRunner_test_EE2Runjob_test.py

```diff
@@ -515,31 +539,66 @@ def test_run_job_batch(self, rq_mock, condor_mock):
         )
         runner = self.getRunner()
         runner.get_condor = MagicMock(return_value=condor_mock)
-        runner.workspace.get_object_info3 = MagicMock(return_value={"paths": []})
+
         job = get_example_job_as_dict(
-            user=self.user_id, wsid=self.ws_id, source_ws_objects=[]
+            user=self.user_id, wsid=None, source_ws_objects=[]
+        )
+        job2 = get_example_job_as_dict(
+            user=self.user_id, wsid=None, source_ws_objects=[]
+        )
+        job3 = get_example_job_as_dict(
+            user=self.user_id, wsid=None, source_ws_objects=[]
         )
         si = SubmissionInfo(clusterid="test", submit=job, error=None)
         condor_mock.run_job = MagicMock(return_value=si)
-        jobs = [job, job, job]
-        job_ids = runner.run_job_batch(params=jobs, batch_params={"wsid": self.ws_id})
+        jobs = [job, job2, job3]
+        job_ids = runner.run_job_batch(
+            params=copy.deepcopy(jobs), batch_params={"wsid": self.ws_id}
+        )
+
+        for job in runner.check_jobs(
+            job_ids=job_ids["child_job_ids"] + [job_ids["parent_job_id"]]
+        )["job_states"]:
+            assert job.get("wsid") == self.ws_id
+            # Job input is forced to assume the batch wsid
+            if job["job_id"] != job_ids["parent_job_id"]:
+                assert job.get("job_input", {}).get("wsid") == self.ws_id
 
         assert "parent_job_id" in job_ids and isinstance(job_ids["parent_job_id"], str)
         assert "child_job_ids" in job_ids and isinstance(job_ids["child_job_ids"], list)
         assert len(job_ids["child_job_ids"]) == len(jobs)
 
+        with self.assertRaises(InvalidParameterForBatch):
+            job_good = get_example_job_as_dict(
+                user=self.user_id, wsid=None, source_ws_objects=[]
+            )
+            job_bad = (
+                get_example_job(user=self.user_id, wsid=self.ws_id).to_mongo().to_dict()
+            )
+            job_bad["method"] = job["job_input"]["method"]
+            job_bad["app_id"] = job["job_input"]["app_id"]
+            job_bad["service_ver"] = job["job_input"]["service_ver"]
```

Comment:
Why don't you need to do this for `job_good` as well?

Reply:
Removed from job_bad

Reply:
:+1: